packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2003,17 +2023,41 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         proposalHandle: string | undefined,
         ackHandle: string,
         summaryLogger: ITelemetryLogger,
+        clientId: string,
     ) {
+        // Tracks whether the summarizer node downloads the snapshot corresponding to the ack.
+        let ackedSnapshotDownloaded = false;
+
+        const getSnapshotP = new LazyPromise<ISnapshotTree>(async () => {
+            ackedSnapshotDownloaded = true;
+            return this.fetchSnapshotFromStorage(ackHandle, summaryLogger, {
+                eventName: "RefreshLatestSummaryGetSnapshot",
+                fetchLatest: false,
+            });
+        });
+
         const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
         await this.summarizerNode.refreshLatestSummary(
             proposalHandle,
-            async () => this.fetchSnapshotFromStorage(ackHandle, summaryLogger, {
-                eventName: "RefreshLatestSummaryGetSnapshot",
-                fetchLatest: false,
-            }),
+            async () => getSnapshotP,
             readAndParseBlob,
             summaryLogger,
         );
+
+        /**
+         * If the acked snapshot was downloaded by the summarizer node and the ack is for a summary that was not
+         * generated by this client, it's possible that it had a different GC version than the current. If so, update
+         * the summary GC version which will update the last successful summary that this client has seen. We will
+         * regenerate the summary and GC data in the next summary to make it up to date to the current version.
+         */
+         if (ackedSnapshotDownloaded && clientId !== this.clientId) {
```

Comment:
I think this code check is totally fine. I am planning to refactor this a bit in a different PR and I will update this part accordingly as well, but I just wanted to let you know the following _should_ be true:
1. All summaries submitted by this summarizer client should be put in SummarizerNode pending summaries
2. When this refresh is called for a summary submitted by this client, it should find the summary in the pending queue, _unless_ it is manually refreshing latest from storage which is also submitted by this client
3. When this refresh is called for a summary _not_ submitted by this client, it will always go down the `refreshLatestFromSnapshot` code path in SummarizerNode and actually download the summary tree, _unless_ it is a summary older than the latest ref seq # we already have.

So all that to say is, theoretically you shouldn't need to do the `clientId !== this.clientId` check, but I wouldn't worry about it for now.

Reply:
That sounds like a good change that will make things clearer. Add me to that PR when it happens so I can get a better idea of this :-).